lib/fetch-template.js

```diff
@@ -25,30 +26,30 @@ class TemplateError extends Error {
  *
  * @param {string} path
  */
-const readFile = (path) => {
-    return new Promise((resolve, reject) => {
+const readFile = path => 
+    new Promise((resolve, reject) => {
         fs.readFile(path, 'utf-8', (err, data) => {
             if (err) {
                 reject(new TemplateError(err));
-                return;
+            } else {
+                resolve(data);
             }
-            resolve(data);
         });
     });
-};
 
 /**
  * Fetches the template from File System
  *
  * @param {string} templatesPath - The path where the templates are stored
  * @param {function=} baseTemplateFn - Function that returns the Base template name for a given page
  */
-module.exports = function fetchTemplate (templatesPath, baseTemplateFn) {
-    return (request, parseTemplate) => {
+module.exports = (templatesPath, baseTemplateFn) => 
+    (request, parseTemplate) => {
```

Comment:
I am keeping parenthesis only when returning object. Since , this function returns a promise there is no need of keeping it. Thoughts?